### ISSUE TITLE
[feature] might you add to save the react state in  'localStorage'?

### ISSUE THREAD

**Issue by Aliaksandr3by** [DmitryVarennikov/cuegenerator-react#25]
[feature] might you add feature  to save the session (the React state) in 'localStorage'?
If the application is closed, all information is reset.

**Comment by DmitryVarennikov** [comment]
Hi @Aliaksandr3by,

What kind of information is lost when the app is closed? Could you explain the use case for a session?

**Comment by Aliaksandr3by** [comment]
? 
the app does not have a state.
all data is erased by closing.
![image](https://user-images.githubusercontent.com/11432724/116787668-19464880-aaae-11eb-8065-65ea8dff875d.png)

**Comment by DmitryVarennikov** [comment]
Right, so what is the use case? Please explain what you need it for so I can understand how the app will benefit from this feature

**Comment by Aliaksandr3by** [comment]
I safe my work, if I close by mistake ?

**Comment by DmitryVarennikov** [comment]
And when you need a clean slate? How do you reset the state?

**Comment by Aliaksandr3by** [comment]
window.localStorage.clear()
or set the variable is ""

**Comment by Aliaksandr3by** [comment]
I think might be beter if ... will be save in localstogare, but it only my wish
Performer:
Title:
File name:
File type:
Tracklist:

**Comment by DmitryVarennikov** [comment]
I appreciate you showing me code examples but my question is rather about UI/UX. 

If app saves its previous state then user (once they filled out the form at least once) will always see the pre-filled page upon return. I.e. this flow leads to the "dirty" form state by default upon return.

**Comment by Aliaksandr3by** [comment]
you might add buttons "load" & "save" & "clear all data";
you might add switcher that will be set save data or not save;
or always save all data, and have only one button "clear data"

Once I used such a code:
But I bad programmer.

`
    constructor(props) {
        super(props);
        this.state = {
            AspNetCoreCookies: this.props.cookies.get(".AspNetCore.Cookies") || "",
            user: {
                login: window.localStorage.getItem("Login") || "",
            },
            error: null,
            isLoaded: false,
            items: [],
            favorite: [],
            favoriteSelect: false,
            favoriteIsLoaded: false,
            filter: this.getItemLocalStorage("filter", []),
            deepSearch: this.getItemLocalStorage("deepSearch", ""),
            pageSize: Number(window.localStorage.getItem("pageSize")) || 10,
            productPage: Number(window.localStorage.getItem("productPage")) || 0,
            totalItems: 0,
            totalPages: 0,
        };

        (async () => {

        })();
    }

    getItemLocalStorage = (name = "", data) => window.localStorage.getItem(name) ? JSON.parse(window.localStorage.getItem(name)) : data;

`

**Comment by DmitryVarennikov** [comment]
It's possible to save the last result locally and ask user if they want to load it. I'm not exactly sure how to ask user, e.g. place a button "Load prev Cue File" next to the "Save Cue to File". 

How often do you need it? You mentioned fixing a mistake, is there any other use-cases?

**Comment by Aliaksandr3by** [comment]
you might save all data in localstorage when User saves CUE.
to give name "tempDateTime"..".tempDateTime.". in localstorage,
and add the "load all" button 

![image](https://user-images.githubusercontent.com/11432724/116791015-469bf200-aac0-11eb-8f0f-6fc1936fd82c.png)

**Comment by Aliaksandr3by** [comment]
if you think what is bad idea, you might close this issue.

**Comment by DmitryVarennikov** [comment]
It may not be a bad idea. While adding new features I'm trying to carefully evaluate it against the following criteria:
1. How useful this feature for **majority** of users and how often it'll be used?
2. Does it block (or annoys, or distracts, etc) anything else?

So you can help me to decide the future! You said you wan to have the previous state when you realize you have made a mistake and the app is already closed. So I have 2 questions:

1. How often does it happen? And what kind of mistake are we talking? if its just one line then guess it's easier to open the downloaded cue file and alter it. 
2. Are there more use-cases other than "oh boy, I made a mistake, lemme redo..."?

**Comment by Aliaksandr3by** [comment]
json converter could get mistaken;
any new feature must arise errors;
I use conv. for only create a cue file, for filling the data I use CALC; I am afraid that I close firefox tab by mistake, and all my work lose.

![image](https://user-images.githubusercontent.com/11432724/116793454-828a8380-aacf-11eb-9dc6-57586e78334f.png)

**Comment by DmitryVarennikov** [comment]
I'm not familiar with your cue-sheet creation flow so I'm not exactly sure what is json converter and how you use it or what is "conv." or "CALC". Can you please clarify those terms and describe your workflow?

**Comment by Aliaksandr3by** [comment]
CALC libreoffice == excel but free);
conv. = convert, for example JSON to STRING, STRING to JSON;
my workflow:
1) I grab timeline from youtube;
2) to modify the timeline by hand in CALC;
3) insert data in your the react APP (why app has name React App?);
4) downloads cue file;
5) slice a single *.flac file into track files;

![image](https://user-images.githubusercontent.com/11432724/116794292-94baf080-aad4-11eb-97ef-010eb763dada.png)

**Comment by DmitryVarennikov** [comment]
Ok I got it. 

The feature _name_ is: **Keep the previous cue form state**
_Description:_ User wants to access previous cue form state in order to make quick amendments to the content (could be useful in case of mistakes/typos). Otherwise user has to re-fill all the fields again to reach the desired state. 

At this point it comes down to the product decision: is this feature worth implementation? There are pros and cons: 
- pros: additional functionality potentially useful for others
- cons: it requires UI change 

To summarize I'd like to solicit more people for their opinions on this feature and put some thoughts into it. I'll come back to you later. And in case I don't just ping me here in 2 weeks.

P.S. It shouldn't be called React app. React is a framework the app is built upon. I filed a [ticket](https://github.com/DmitryVarennikov/cuegenerator-react/issues/27) to fix it.

**Comment by DmitryVarennikov** [comment]
@Aliaksandr3by I followed your advice and implemented previous state button. Once you hit the "Save cue" button it'll also remember this cue and you'll see another button called "Load my prev Cue". Now you can refresh your browser page, hit load cue and enjoy

┊┊ ☆┊┊┊┊☆┊┊☆ ┊┊┊┊┊
┈┈┈┈╭━━━━━━╮┊☆ ┊┊
┈☆ ┈┈┃╳╳╳▕╲▂▂╱▏┊┊
┈┈☆ ┈┃╳╳╳▕▏▍▕▍▏┊┊
┈┈╰━┫╳╳╳▕▏╰┻╯▏┊┊
☆ ┈┈┈┃╳╳╳╳╲▂▂╱┊┊┊
┊┊☆┊╰┳┳━━┳┳╯┊ ┊ ☆┊
